core/app/models/spree/stock/estimator.rb

```diff
@@ -3,9 +3,14 @@
 module Spree
   module Stock
     class Estimator
+      attr_reader :order
       class ShipmentRequired < StandardError; end
       class OrderRequired < StandardError; end
 
+      def initialize(order = nil)
+        @order = order
```

Comment:
Should we print a deprecation warning telling people that initializing the estimator without `order` is not going to be supported in the next major?

Reply:
I'll add a deprecation message here, I couldn't find any documentation that specifies how to add your own custom estimator so the usage is probably low.

Reply:
@HendrikLouw overriding the stock estimator is pretty common practice for stores with advanced stock logic. Some extensions even provide their own estimator, e.g. [solidus_easypost](https://github.com/solidusio-contrib/solidus_easypost/blob/master/lib/solidus_easypost/estimator.rb) does it.

Reply:
Got it, thank you for the information, I've added a deprecation message now and that is highlighting that a lot of our tests are not passing in the `order` to the initialiser.  Do you think we should update the tests as part of this change?

Reply:
@aldesantis I see CI doesn't allow the deprecation message, I'll fix this up tonight.